src/main.c

```diff
@@ -0,0 +1,34 @@
+#include <zephyr.h>
+#include <stdio.h>
+#include <sys/printk.h>
+extern void send_main(void);
+extern void receive_main(void);
+
+extern void rtc_main(void);
+
+extern void slave_setup_main(void);
+extern void i2c_master_main(void);
+
+
+void main(void)
+{
+
+    #ifdef SEND_ACTIVITY
+        send_main();
+    #endif
+
+    #ifdef RECEIVE_ACTIVITY
+        receive_main();
+    #endif
+
+    #ifdef RTC_TEST
+        rtc_main();
```

Comment:
rtc_main has a forever while loop, so this (and maybe others) would need to be in their own threads.

Reply:
resolved with the while loops in the setup so this should run without needing to be in their own thread